### PR TITLE
Debug display mostly OK

### PR DESCRIPTION
**Pull request by louist103**
Before opening this PR, ensure the following:
- `./format.sh` was run to apply standard formatting.
- `make` successfully builds a matching ROM.
- No new compiler warnings were introduced during the build process.
    - Can be verified locally by running `tools/warnings_count/check_new_warnings.sh`
- New variables & functions should follow standard naming conventions.
- Comments and variables have correct spelling.
---

Despite the name of the branch this is Debug display.  Wasn't sure what to do about the lights, and the last function. All decompiled function match. Most documentation was taken from OOT.

### CODE REVIEW

File: src/code/z_debug_display.c [hunk 1]
```diff
@@ -1,15 +1,150 @@
...
+    sDebugObjectListHead->color.r = red;
+    sDebugObjectListHead->color.g = green;
+    sDebugObjectListHead->color.b = blue;
+    sDebugObjectListHead->color.a = alpha;
+    sDebugObjectListHead->type = type;
+    sDebugObjectListHead->next = oldHead;
+    return sDebugObjectListHead;
+}
+
+#include "code/debug_display/code.c"
```
Comment (AngheloAlf) [inline src/code/z_debug_display.c:44]: This is a bit weird
You can put `OutName="debug_display"` in the XML. That way ZAPD will generate a `code/debug_display/debug_display.c` file instead

File: src/code/z_debug_display.c [hunk 2]
```diff
@@ -1,15 +1,150 @@
...
+    while (dispObj != NULL) {
+        objInfo = &sDebugObjectInfoTable[dispObj->type];
+        sDebugObjectDrawFuncTable[objInfo->drawType](dispObj, objInfo->drawArg, globalCtx);
+        dispObj = dispObj->next;
+    }
+}
+
+void DebugDisplay_DrawSpriteI8(DebugDispObject* dispObj, void* texture, GlobalContext* globalCtx) {
+
+    OPEN_DISPS(globalCtx->state.gfxCtx);
```
Comment (AngheloAlf) [inline src/code/z_debug_display.c:67]: You can remove this empty line
```suggestion
void DebugDisplay_DrawSpriteI8(DebugDispObject* dispObj, void* texture, GlobalContext* globalCtx) {
    OPEN_DISPS(globalCtx->state.gfxCtx);
```

File: src/code/z_debug_display.c [hunk 3]
```diff
@@ -1,15 +1,150 @@
...
+s32 sDebugDisplayLight2[] = {
+    0xFFFFFF00,
+    0xFFFFFF00,
+    0x49494900,
+    0x00000000,
+};
+
+void DebugDisplay_DrawPolygon(DebugDispObject* dispObj, void* arg1, GlobalContext* globalCtx) {
+
+    OPEN_DISPS(globalCtx->state.gfxCtx);
```
Comment (AngheloAlf) [inline src/code/z_debug_display.c:111]:
```suggestion
void DebugDisplay_DrawPolygon(DebugDispObject* dispObj, void* arg1, GlobalContext* globalCtx) {
    OPEN_DISPS(globalCtx->state.gfxCtx);
```

File: src/code/z_debug_display.c [hunk 4]
```diff
@@ -1,15 +1,150 @@
...
+    gDPLoadSync(POLY_XLU_DISP++);
+
+    gDPLoadBlock(POLY_XLU_DISP++, G_TX_LOADTILE, 0, 0, 127, 1024);
+
+    gDPPipeSync(POLY_XLU_DISP++);
+
+    gDPSetTile(POLY_XLU_DISP++, G_IM_FMT_I, G_IM_SIZ_8b, 2, 0x0000, G_TX_RENDERTILE, 0, G_TX_NOMIRROR | G_TX_WRAP,
+               G_TX_NOMASK, G_TX_NOLOD, G_TX_NOMIRROR | G_TX_WRAP, G_TX_NOMASK, G_TX_NOLOD);
+
+    gDPSetTileSize(POLY_XLU_DISP++, G_TX_RENDERTILE, 0, 0, 0x003C, 0x003C);
```
Comment (AngheloAlf) [inline src/code/z_debug_display.c:88]: If I'm not wrong, all of this is just simply this:
```suggestion
    gDPLoadTextureBlock(POLY_XLU_DISP++, texture, G_IM_FMT_I, G_IM_SIZ_8b, 16, 16, 0, G_TX_NOMIRROR | G_TX_WRAP, G_TX_NOMIRROR | G_TX_WRAP, G_TX_NOMASK, G_TX_NOMASK, G_TX_NOLOD, G_TX_NOLOD);
```

**Review by AngheloAlf — COMMENTED** [review]
A few things

File: src/code/z_lights.c [hunk 5]
```diff
@@ -329,8 +329,7 @@ Lights* Lights_NewAndDraw(GraphicsContext* gfxCtx, u8 ambientR, u8 ambientG, u8
     Lights* lights;
     s32 i;
 
-    // TODO allocation should be a macro
-    lights = (Lights*)((int)gfxCtx->polyOpa.d - sizeof(Lights));
+    lights = GRAPH_ALLOC(gfxCtx, sizeof(Lights));
     gfxCtx->polyOpa.d = (void*)lights;
```
Comment (hensldm) [inline src/code/z_lights.c:333]:
```suggestion
```
Should be able to remove this as it is a part of the macro.

File: src/code/z_lights.c [hunk 6]
```diff
@@ -356,8 +355,7 @@ Lights* Lights_NewAndDraw(GraphicsContext* gfxCtx, u8 ambientR, u8 ambientG, u8
 Lights* Lights_New(GraphicsContext* gfxCtx, u8 ambientR, u8 ambientG, u8 ambientB) {
     Lights* lights;
 
-    // TODO allocation should be a macro
-    lights = (Lights*)((int)gfxCtx->polyOpa.d - sizeof(Lights));
+    lights = GRAPH_ALLOC(gfxCtx, sizeof(Lights));
     gfxCtx->polyOpa.d = (void*)lights;
```
Comment (hensldm) [inline src/code/z_lights.c:359]:
```suggestion
```
Same here

File: src/code/z_skin_matrix.c [hunk 7]
```diff
@@ -577,11 +577,8 @@ void SkinMatrix_MtxFToMtx(MtxF* src, Mtx* dest) {
 }
 
 Mtx* SkinMatrix_MtxFToNewMtx(GraphicsContext* gfxCtx, MtxF* src) {
-    s32 pad;
-    Mtx* mtx;
+    Mtx* mtx = GRAPH_ALLOC(gfxCtx, sizeof(Mtx));
 
-    // TODO allocation should be a macro
-    mtx = (Mtx*)((int)gfxCtx->polyOpa.d - sizeof(Mtx));
     gfxCtx->polyOpa.d = (void*)mtx;
```
Comment (hensldm) [inline src/code/z_skin_matrix.c:582]:
```suggestion
```
Same here

**Review by hensldm — CHANGES_REQUESTED** [review]
A few things with the GRAPH_ALLOC macro.

**Review by AngheloAlf — APPROVED** [review]

**Review by Kenix3 — APPROVED** [review]
Merge pending +1 contributor approval.
Please remember to resolve comments you have addressed, and comment on ones you won't then resolve.

File: src/code/z_lights.c (hunk 5 above)
Comment (louist103) [reply src/code/z_lights.c:333]: This creates warnings. Same for other examples.

Comment (hensldm) [reply src/code/z_lights.c:333]: Out of curiosity, could you post the warnings?

**Review by hensldm — APPROVED** [review]
LGTM

**Pull request merged**